src/graph.py

```diff
@@ -0,0 +1,122 @@
+"""A gragh data structure."""
+
+
+class Graph(object):
+    """Graph."""
+
+    def __init__(self):
+        """Create the dictionary that will be the graph."""
+        self.g = {}
+
+    def nodes(self):
+        """Return a list of all the nodes in the graph."""
+        nodes = []
+        for key in self.g:
+            nodes += [key]
+        return nodes
+
+    def edges(self):
+        """Return a list of all the edges in the graph."""
+        edges = []
+        for key in self.g:
+            for edge in self.g[key]:
+                edges += [(key, edge)]
+        return edges
+
+    def add_node(self, n):
+        """Add a node to the graph."""
+        self.g.setdefault(n, [])
+
+    def add_edge(self, n1, n2):
```

Comment:
This comment on the code has gone entirely unaddressed, and the method for implementing weights on a graph's edges below is improperly implemented. The weight should be able to be user supplied. Please resubmit.